amundsen_application/api/utils/request_utils.py

```diff
@@ -55,18 +60,20 @@ def request_search(*,     # type: ignore
                            url=url,
                            client=app.config['SEARCHSERVICE_REQUEST_CLIENT'],
                            headers=headers,
-                           timeout_sec=timeout_sec)
+                           timeout_sec=timeout_sec,
+                           json=json)
 
 
 # TODO: Define an interface for envoy_client
-def request_wrapper(method: str, url: str, client, headers, timeout_sec: int):  # type: ignore
+def request_wrapper(method: str, url: str, client, headers, timeout_sec: int, json: str = ''):  # type: ignore
```

Comment:
I'm not super familiar with python, but would there be any unintended consequences from using an empty string as a default value? @feng-tao

Reply:
From the metadata PR, this value gets used by `json.loads()`. @Mikhail-Ivanov what is the result of using `json.loads()` on an empty string? 

Tried looking this up myself but I couldn't quite find some clear documentation on what the default behavior is.

Reply:
I think empty string is fine and keep it as it is will help to make it backward compatible.

Reply:
How so? 

As mentioned, I see the[ empty string gets passed](https://github.com/lyft/amundsenmetadatalibrary/pull/79/files#diff-4b1a079f306204ef1278fb5acc3a4a23R30) to `json.loads()` and is not by itself the default description. I think we need to confirm what `json.loads("")` returns -- is it `None`, `{}`, or some error?

Reply:
@ttannis, @feng-tao here are my findings:
https://github.com/lyft/amundsenfrontendlibrary/pull/298#discussion_r327478289
https://github.com/lyft/amundsenmetadatalibrary/pull/79#discussion_r327473924
Looks like the only way to truncate description is to intentionally send an empty "description" key

Reply:
From your comment [here](https://github.com/lyft/amundsenfrontendlibrary/pull/298#discussion_r327478289) it looks like `json.loads('')` threw an error `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)` whereas `json.loads('{}')` resulted in an empty dictionary `{}`. 

Considering `''` vs. `'{}'` as the default encoded json string, while they both result in the description not being updated -- which is what we want -- I am leaning towards `'{}'` as the default encoded string. I think it is more correct for the default to resolve to an empty dictionary as opposed to choosing a default that is guaranteed to throw an error if it gets used.

Reply:
@ttannis  are we ok to use mutable as default argument given it is a bad practice in python world(https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)?

Reply:
This one isn't mutable because the default is not an empty dictionary itself, but an encoded json string that will resolve to a new empty dictionary each time. 

Here's some code I just tried:
```
import json

# when the default is an encoded json string that has to be parsed by json.loads()
def this_is_safe(key, value, j = '{}'):
    test_dict = json.loads(j)
    test_dict[key] = value
    return test_dict

print(this_is_safe('test', 'testval'))     # {'test': 'testval'}
print(this_is_safe('test2', 'testval2'))   # {'test2': 'testval2'}


# when the default is an actual dictionary
def this_is_not_safe(key, value, j = {}):
    test_dict = j
    test_dict[key] = value
    return test_dict

print(this_is_not_safe('test', 'testval'))    # {'test': 'testval'}
print(this_is_not_safe('test2', 'testval2'))  # {'test': 'testval', 'test2': 'testval2'}
```

Reply:
oh, initially I thought the default is `{}` not `'{}'`, if it is former case, yeah, it won't be mutable per se.

Reply:
@ttannis 
> I am leaning towards `'{}'` as the default encoded string
Agree, it's a better option. Done